### PR TITLE
Add fields_meta and an upgrade script to move existing editor_state

### PR DESCRIPTION
**Pull request by tomaskikutis**
SDFID-345

MUST BE MERGED TOGETHER WITH CLIENT CHANGES https://github.com/superdesk/superdesk-client-core/pull/2180

### CODE REVIEW

File: superdesk/metadata/item.py [hunk 1]
```diff
@@ -524,11 +524,11 @@
...
 
-    # draft-js editor state
-    'editor_state': {
-        'type': 'list',
-        'nullable': True,
-        'mapping': not_enabled,
+    'fields_meta': {
+        'type': 'dict',
+        'mapping': {
+            'type': 'object',
```
Comment (petrjasek) [inline superdesk/metadata/item.py]: can you specify the properties? it would be better to avoid any dynamic mapping

Comment (tomaskikutis) [reply superdesk/metadata/item.py]: It can contain custom fields created by the user, so it's not possible to have a static mapping

Comment (petrjasek) [reply superdesk/metadata/item.py]: can you set it to `not_enabled` then? it shouldn't get indexed

**Comment by tomaskikutis** [comment]
Closed in favor of #1238

**Pull request closed without merging**